src/snowflake/connector/config_manager.py

```diff
@@ -320,8 +320,15 @@ def read_config(
         ):
             if sliceoptions.only_in_slice:
                 del read_config_file[section]
-            if not filep.exists():
+            try:
+                if not filep.exists():
+                    continue
+            except PermissionError:
+                LOGGER.debug(
```

Comment:
Why did you add a `debug` log for failing to read a configuration file.
I do not think that's an appropriate logging level for such an case.